webapp/src/types/mattermost-webapp/index.d.ts

```diff
@@ -76,3 +78,46 @@ export type SurveyResult = {
     responseCount: number;
     status: SurveyStatus;
 }
+
+export type FormatTextOptions = {
+    markdown?: boolean;
+}
+
+export type PostUtils = {
+    formatText: (text: string, option: Partial<FormatTextOptions>) => string;
+    messageHtmlToComponent: (html: string, isRHS: boolean) => React.ReactNode;
+}
+
+export type MattermostWindow = {
+    PostUtils: PostUtils;
+}
+
+export type CustomPostTypeComponentProps = {
+    post: Post;
+    isRHS: boolean;
+}
+
+export type SurveyResponse = {
+    responses: {[key: string]: unknown};
+    dateCreated: string;
+}
+
+export type Survey = {
+    surveyId: string;
+    startDate: string;
+    endDate: string;
+    questions: Question[];
+    status: SurveyStatus;
+}
+
+export type Question = {
+    id: string;
+    text?: string;
+    type: QuestionType;
+    system: boolean;
+    mandatory: boolean;
+};
```

Comment:
This file should define types coming from the `mattermost-webapp` package, isn't it? These don't seem to be part of the mattermost webapp package. Shouldn't they live in a different file?

Reply:
Ah yes, you're right, its under the mm webapp dir. I've created a new `plugin.d.ts` file now and moved plugin specific types in there.